gallery/v2_transforms/plot_transforms_v2.py

```diff
@@ -7,103 +7,253 @@
     Try on `collab <https://colab.research.google.com/github/pytorch/vision/blob/gh-pages/main/_generated_ipynb_notebooks/plot_transforms_v2.ipynb>`_
     or :ref:`go to the end <sphx_glr_download_auto_examples_v2_transforms_plot_transforms_v2.py>` to download the full example code.
 
-Most computer vision tasks are not supported out of the box by ``torchvision.transforms`` v1, since it only supports
-images. ``torchvision.transforms.v2`` enables jointly transforming images, videos, bounding boxes, and masks. This
-example showcases the core functionality of the new ``torchvision.transforms.v2`` API.
+This example illustrates all of what you need to know to get started with the
+new :mod:`torchvision.transforms.v2` API. We'll cover simple tasks like
+image classification, and more advanced ones like object detection /
+segmentation.
 """
 
-import pathlib
-
+# %%
+# First, a bit of setup
+from pathlib import Path
 import torch
+import matplotlib.pyplot as plt
+plt.rcParams["savefig.bbox"] = 'tight'
 
+from torchvision.transforms import v2
+from torchvision.io import read_image
 
-def load_data():
-    from torchvision.io import read_image
-    from torchvision import datapoints
-    from torchvision.ops import masks_to_boxes
+torch.manual_seed(1)
 
-    assets_directory = pathlib.Path("../assets")
+# If you're trying to run that on collab, you can download the assets and the
+# helpers from https://github.com/pytorch/vision/tree/main/gallery/
+from helpers import plot
+img = read_image(str(Path('../assets') / 'astronaut.jpg'))
+print(f"{type(img) = }, {img.dtype = }, {img.shape = }")
 
-    path = assets_directory / "FudanPed00054.png"
-    image = datapoints.Image(read_image(str(path)))
-    merged_masks = read_image(str(assets_directory / "FudanPed00054_mask.png"))
+# %%
+# The basics
+# ----------
+#
+# The Torchvision transforms behave like a regular :class:`torch.nn.Module` (in
+# fact, most of them are): instantiate a transform, pass an input, get a
+# transformed output:
 
-    labels = torch.unique(merged_masks)[1:]
+transform = v2.RandomCrop(size=(224, 224))
+out = transform(img)
 
-    masks = datapoints.Mask(merged_masks == labels.view(-1, 1, 1))
+plot([img, out])
 
-    bounding_boxes = datapoints.BoundingBoxes(
-        masks_to_boxes(masks), format=datapoints.BoundingBoxFormat.XYXY, canvas_size=image.shape[-2:]
-    )
+# %%
+# I just want to do image classification
+# --------------------------------------
+#
+# If you just care about image classification, things are very simple. A basic
+# classification pipeline may look like this:
 
-    return path, image, bounding_boxes, masks, labels
+transforms = v2.Compose([
```

Comment:
Should we avoid the plural for the variable name to not clash with the namespace? Somewhat weak argument, since for some datasets we would have `transforms=transform`. Still, I would prefer keeping it singular here and in general.

Reply:
This is sort of related to https://github.com/pytorch/vision/issues/7866.

These are transforms, it makes sense to call this variable transforms. If there was a single transform I'd call it transform, or t.

I personally strongly dislike the new trend of important the module as `transforms` in our tests.